File: src/TheWorld/wwwroot/js/tripsController.js
(function () {
    "use strict";

    // Getting the existing modules
    angular.module("app-trips")
    .controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;
        vm.trips = [];

        vm.newTrip = {};
        vm.errorMessage = "";
        vm.isBusy = true;
        $http.get("/api/trips")
            .then(function (response) {
                angular.copy(response.data, vm.trips);
            }, function (error) {
                //failure
                vm.errorMessage = "Failed to load data " + error;
                vm.isBusy = false;
            })
        .finally(function () {
            vm.isBusy = false;
        });

        vm.addTrip = function () {
            vm.isBusy = true;
            vm.errorMessage = "";

            $http.post("/api/trips", vm.newTrip)
            .then(function (response) {
                vm.trips.push(response.data);
            }, function () {
                vm.errorMessage = "Failed to save new trip";
            })
            .finally(function() {
                vm.isBusy = false;
            });
        };
    }
})();

File: src/TheWorld/Controllers/Api/TripsController.cs
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using TheWorld.Models;

namespace TheWorld.Controllers.Api
{
    [Authorize]
    [Route("/api/trips")]
    public class TripsController : Controller
    {
        private ILogger<TripsController> _logger;
        private IWorldRepository _repository;

        public TripsController(IWorldRepository repository, 
            ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }

        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {

                //var trips = _repository.GetAllTripsWithStops();
                var trips = _repository.GetUserTripsWithStops(User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(trips));
            }
            catch (Exception ex)
            {
                // TODO Logging;
                _logger.LogError($"Failed to get all Trops: {ex}");
                return BadRequest("Error occured");
            }
        }

        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel vm)
        {
            if (ModelState.IsValid)
            {
                var newTrip = Mapper.Map<Trip>(vm);
                newTrip.UserName = User.Identity.Name;

                _logger.LogInformation("Attempting to save new trip");
                _repository.AddTrip(newTrip);

                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{vm.Name}", 
                        Mapper.Map<TripViewModel>(newTrip));
                }
            }
            return BadRequest("Failed to save the trip");
        }
    }
}
